packages/@sanity/form-builder/src/inputs/ImageInput/Asset.js

```diff
@@ -0,0 +1,254 @@
+
+  renderMenuItem = item => {
+    const {color, title, icon} = item
+    const Icon = icon
+    return (
+      <div className={color === 'danger' ? styles.menuItemDanger : styles.menuItem}>
+        {icon && <Icon />}&nbsp;&nbsp;{title}
+      </div>
+    )
+  }
+
+  getDialogActions = dialogType => {
+    const actions = []
+
+    if (dialogType != 'cantDelete') {
+      actions.push({name: 'delete', title: 'Delete', icon: TrashIcon, color: 'danger'})
+    }
+
+    actions.push({
+      name: 'close',
+      title: 'Close'
+    })
+
+    return actions
+  }
+
+  render() {
+    const {asset, onClick, onKeyPress} = this.props
+    const {isDeleting, showDialog, dialogType} = this.state
+    const size = 75
+    const width = get(asset, 'metadata.dimensions.width') || 100
+    const height = get(asset, 'metadata.dimensions.height') || 100
+
+    const menuItems = [
+      {
+        name: 'showRefs',
+        title: 'Show documents using this',
+        icon: LinkIcon
+      },
+      {
+        name: 'delete',
+        title: 'Delete',
+        color: 'danger',
+        icon: TrashIcon
+      }
+    ]
+
+    return (
+      <a
+        className={styles.item}
+        tabIndex={0}
+        data-id={asset._id}
+        onKeyPress={onKeyPress}
+        style={{
+          width: `${(width * size) / height}px`,
+          flexGrow: `${(width * size) / height}`
+        }}
+      >
+        <i className={styles.padder} style={{paddingBottom: `${(height / width) * 100}%`}} />
+        {/* We can not determine an alt text on image */}
+        <img
+          src={`${asset.url}?h=100`}
+          className={styles.image}
+          onClick={onClick}
+          data-id={asset._id}
+        />
+        {isDeleting && (
+          <div className={styles.spinnerContainer}>
+            <Spinner center />
+          </div>
+        )}
+        <div className={styles.menuContainer}>
+          <DropDownButton
+            placement="bottom-end"
+            showArrow={false}
+            items={menuItems}
+            renderItem={this.renderMenuItem}
+            onAction={this.handleMenuAction}
+          >
+            <MoreVertIcon />
+          </DropDownButton>
+          {showDialog && (
+            <Dialog
+              title={
+                dialogType === 'cantDelete' ? 'Could not delete asset' : 'Documents using this'
+              }
+              color={dialogType === 'cantDelete' ? 'danger' : undefined}
+              onClose={this.handleDialogClose}
+              onAction={this.handleDialogAction}
+              actions={this.getDialogActions(dialogType)}
+            >
+              <DialogContent size="medium">
+                <div className={styles.dialogContent}>
+                  <img src={`${asset.url}?w=200`} style={{maxWidth: '200px'}} />
+                  <WithReferringDocuments id={asset._id}>
+                    {({isLoading, referringDocuments}) => {
+
+                      const drafts = referringDocuments.reduce(
```

Comment:
Could this array be named `documentIdsWithDrafts` or something for clarity? Had to read this block several times to get what was going on :) Would also be good to use the functions from the part `part:@sanity/base/util/draft-utils` in order to rely on the same logic for dealing with ids of drafts and their published counterpart.

Check out `./packages/@sanity/base/src/util/draftUtils.js` (which is the one implemente by the part mentioned above)